src/curry/once.ts

```diff
@@ -0,0 +1,55 @@
+const onceResult = '__once__'
+
+/**
+ * The type of a function wrapped with `once`.
+ */
+export interface OnceFunction<
+  Args extends unknown[] = unknown[],
+  Return = unknown,
+  This = unknown,
+> {
+  (this: This, ...args: Args): Return
+  [onceResult]?: Return
+}
+
+/**
+ * Create a function that runs at most once, no matter how many times
+ * it's called. If it was already called before, returns the result
+ * from the first call. This is a lighter version of `memo()`.
+ *
+ * To allow the function to be called again, use `onceReset()`.
+ *
+ * ```ts
+ * const fn = once(() => Math.random())
+ * fn() // 0.5
+ * fn() // 0.5
+ * ```
+ */
+export function once<Args extends unknown[], Return, This = unknown>(
+  fn: (this: This, ...args: Args) => Return,
+): (this: This, ...args: Args) => Return {
+  const onceFn = function (this, ...args) {
```

Comment:
Can you please add a test to check that we pass `this` correctly into `fn`

Reply:
Good catch!